course-materials/lab-instructions/lab-06/lab-06-ugly-charts.Rmd

```diff
@@ -48,35 +48,96 @@ use_git_config(user.name = "Your Name",
 
 This is the second week you're working in teams, so we're going to make things a little more interesting and let all of you make changes and push those changes to your team repository. Sometimes things will go swimmingly, and sometimes you'll run into merge conflicts. So our first task today is to walk you through a merge conflict!
 
-## Merge conflicts
 
-- When two collaborators make changes to a file and push the file to their repo, git merges these two files.
-- If these two files have conflicting content on the same line, git will produce a **merge conflict**.
+## Merges and merge conflicts
 
-### Set up
+- Pushing a repo replaces the code on GitHub with the code you have on your computer. 
+- If someone else has made a change that you haven't incorporated into your repo, GitHub will stop you from pushing because it would lose their work!
+- So you need to explicitly "merge" your collaborators' work before you push.
+- If the changes are in different files, or different parts of the same file,
+  git merges the work for you automatically when you *pull*.
+- If you both changed the same part of a file, git will produce a **merge conflict**
+  because it doesn't know how which change to take.
+  
+Git will put conflict markers in your code that look like:
 
-- Clone the repo for your next assignment in RStudio, and open the .Rmd file.
-- Assign the numbers 1, 2, 3, and 4 to each of the team members.
+```
+<<<<<<< HEAD
+See also: [re docs](https://docs.python.org/3/library/re.html)  
+=======
+See also [re docs](https://docs.python.org/3/library/re.html)  
+>>>>>>> master
```

Comment:
```suggestion
<<<<<<< HEAD
See also: [dplyr documentation](https://dplyr.tidyverse.org/)  
=======
See also [ggplot2 documentation](https://ggplot2.tidyverse.org/)  
>>>>>>> some1alpha2numeric3string4
```

Reply:
I made the difference more obvious and also R related (just so we reduce the number of unfamiliar things they see within this activity).

Also, it shouldn't be `master` but a hash for the last commit so made an "informative" text string but I wonder if an actual hash is better...